src/charm.py

```diff
@@ -7,26 +7,45 @@
 import yaml
 
 from charmhelpers.core import hookenv
+from charms.loki_k8s.v0.loki_push_api import LokiPushApiConsumer
 from ops.charm import CharmBase
 from ops.main import main
-from ops.model import ActiveStatus, BlockedStatus
+from ops.model import ActiveStatus, BlockedStatus, MaintenanceStatus
 from ops.framework import StoredState
+from subprocess import check_call
 
 logger = logging.getLogger(__name__)
 
+LOGGING_RELATION_NAME = "logging"
+
 
 class JujuControllerCharm(CharmBase):
     _stored = StoredState()
 
     def __init__(self, *args):
         super().__init__(*args)
+        self.framework.observe(self.on.install, self._on_install)
```

Comment:
in theory, the remove hook would remove promtail but we do disallow removal of the controller charm so not needed
maybe add a comment to that effect